Add git sync

colonelpanic8 opened the pull request:
###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

This is my the first derivation I've added, so let me know if anything looks strange.

I'm sure this is explained somewhere, but its not clear to me how useSandbox should be used, or where it should be set. The build section of this package is literally just a copy, but i'd still like to know how to do this.

colonelpanic8 committed:
Add imalison maintainer

colonelpanic8 committed:
git-sync: init at 20151024

ryantm commented on pkgs/tools/backup/git-sync/default.nix:
```diff
@@ -0,0 +1,44 @@
+}:
+
+stdenv.mkDerivation rec {
+  name = "git-sync";
```
The version should be part of the name like `"name-${version}"`. In the case of a project like this with no releases, we typically use the date the commit was made Oct 24, 2015, so we'd use `20151024` as the version. This would let us have increasing version numbers when we make updates to the package. (Assuming updates are continuing for this package!)

jtojnar replied:
The version should probably be `unstable-2015-10-24`, see https://nixos.org/nixpkgs/manual/#sec-package-naming.

IvanMalison commented:
@ryantm Is there a better way to test the resulting binary than using

```
nix-build -A git-sync -K
```

And then testing what is produced in the result directory?

xeji commented:
This package would better fit in the `pkgs/applications/version-management/git-and-tools` directory, which contains all git related tools.

ryantm commented:
@IvanMalison You could build it in a more restricted way:

```
nix-build --option sandbox true --option restrict-eval true -A git-sync -K
```

Can you change the commit message to say "git-sync: init at 20151024"?

I think xeji is right about the file location too.

ryantm commented:
@GrahamcOfBorg build git-sync

GrahamcOfBorg commented:
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.41876&attempt_id=88732ed8-7499-4e59-8d78-590383cd1273)

Attempted: git-sync

<details><summary>Partial log (click to expand)</summary><p>

```
source root is source
patching sources
updateAutotoolsGnuConfigScriptsPhase
configuring
no configure script, doing nothing
installing
post-installation fixup
patching script interpreter paths in /nix/store/zkpw89dxkkwp5dwyvjazv1qx2df2qsjw-git-sync-20151024/bin
/nix/store/zkpw89dxkkwp5dwyvjazv1qx2df2qsjw-git-sync-20151024/bin/git-sync: interpreter directive changed from "/usr/bin/env bash" to "/nix/store/9rv2bzx7jz6wjbaxij5qq580qvzksj4i-bash-4.4-p19/bin/bash"
/nix/store/zkpw89dxkkwp5dwyvjazv1qx2df2qsjw-git-sync-20151024
```
</p></details>

GrahamcOfBorg commented:
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.41876&attempt_id=02ce3ed8-e8b6-4915-8ef4-01b12c12a28f)

Attempted: git-sync

<details><summary>Partial log (click to expand)</summary><p>

```
unpacking source archive /nix/store/mq5m03kfalv5my77934l8gqz1v0hc8zd-source
source root is source
patching sources
configuring
no configure script, doing nothing
installing
post-installation fixup
patching script interpreter paths in /nix/store/ywasxkp7jg4yxnai6kwdks7w3jx6l9yz-git-sync-20151024/bin
/nix/store/ywasxkp7jg4yxnai6kwdks7w3jx6l9yz-git-sync-20151024/bin/git-sync: interpreter directive changed from "/usr/bin/env bash" to "/nix/store/qckzjk3406va7h6s40cy9s75z2w715rq-bash-4.4-p19/bin/bash"
/nix/store/ywasxkp7jg4yxnai6kwdks7w3jx6l9yz-git-sync-20151024
```
</p></details>

GrahamcOfBorg commented:
Success on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.41876&attempt_id=f309e975-de78-4c2f-9e11-86fa3c8684f3)

Attempted: git-sync

<details><summary>Partial log (click to expand)</summary><p>

```
unpacking source archive /nix/store/mq5m03kfalv5my77934l8gqz1v0hc8zd-source
source root is source
patching sources
configuring
no configure script, doing nothing
installing
post-installation fixup
patching script interpreter paths in /nix/store/bzflg4s6myvqhvdxf2qwfh4i2nls4z7h-git-sync-20151024/bin
/nix/store/bzflg4s6myvqhvdxf2qwfh4i2nls4z7h-git-sync-20151024/bin/git-sync: interpreter directive changed from "/usr/bin/env bash" to "/nix/store/9gghkkqhn6v7dvpf2319qvrxkkw03vwr-bash-4.4-p19/bin/bash"
/nix/store/bzflg4s6myvqhvdxf2qwfh4i2nls4z7h-git-sync-20151024
```
</p></details>

IvanMalison commented:
@ryantm I think that in this case, sinceall building involves copying a file, the more important test is at runtime.

I tried running the resulting wrapped binary with `PATH=""` and things seemed to work.
Is there are more nixosy way to do this?

IvanMalison commented:
@ryantm Things look good.

Is something blocking merge?

xeji committed:
git-sync: move to gitAndTools.git-sync

By convention, git-related tools in nixpkgs are not added to
the top level but bundled under gitAndTools.

xeji commented:
@GrahamcOfBorg  build gitAndTools.git-sync

GrahamcOfBorg commented:
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.41876&attempt_id=4a73bb12-7a6e-4f94-945d-4702dd4056dd)

Attempted: gitAndTools.git-sync

<details><summary>Partial log (click to expand)</summary><p>

```
source root is source
patching sources
updateAutotoolsGnuConfigScriptsPhase
configuring
no configure script, doing nothing
installing
post-installation fixup
patching script interpreter paths in /nix/store/5102bs0fk2m0knzshr3iacbk2kc3fyqn-git-sync-20151024/bin
/nix/store/5102bs0fk2m0knzshr3iacbk2kc3fyqn-git-sync-20151024/bin/git-sync: interpreter directive changed from "/usr/bin/env bash" to "/nix/store/p0vy17dp9jk2mvqsxsqnb14s3797lay7-bash-4.4-p23/bin/bash"
/nix/store/5102bs0fk2m0knzshr3iacbk2kc3fyqn-git-sync-20151024
```
</p></details>

GrahamcOfBorg commented:
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.41876&attempt_id=cf546a94-e52d-4e12-87f7-f5960c78da40)

Attempted: gitAndTools.git-sync

<details><summary>Partial log (click to expand)</summary><p>

```
unpacking source archive /nix/store/mq5m03kfalv5my77934l8gqz1v0hc8zd-source
source root is source
patching sources
configuring
no configure script, doing nothing
installing
post-installation fixup
patching script interpreter paths in /nix/store/l8n3azg54lv31i8c5nbwn0s18dzwn47p-git-sync-20151024/bin
/nix/store/l8n3azg54lv31i8c5nbwn0s18dzwn47p-git-sync-20151024/bin/git-sync: interpreter directive changed from "/usr/bin/env bash" to "/nix/store/8zkg9ac4s4alzyf4a8kfrig1j73z66dw-bash-4.4-p23/bin/bash"
/nix/store/l8n3azg54lv31i8c5nbwn0s18dzwn47p-git-sync-20151024
```
</p></details>

xeji commented:
@IvanMalison By convention, git-related tools in nixpkgs are not added to
the top level but bundled under gitAndTools. I added a commit to move it there.
It can now be installed as `gitAndTools.git-sync`.

GrahamcOfBorg commented:
Success on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.41876&attempt_id=8ed9a43a-f4f1-4a78-8528-c14841a1d1fa)

Attempted: gitAndTools.git-sync

<details><summary>Partial log (click to expand)</summary><p>

```
unpacking source archive /nix/store/mq5m03kfalv5my77934l8gqz1v0hc8zd-source
source root is source
patching sources
configuring
no configure script, doing nothing
installing
post-installation fixup
patching script interpreter paths in /nix/store/ki845cvn6garxi6l7bdwjj3msm0hfzg5-git-sync-20151024/bin
/nix/store/ki845cvn6garxi6l7bdwjj3msm0hfzg5-git-sync-20151024/bin/git-sync: interpreter directive changed from "/usr/bin/env bash" to "/nix/store/q2wqq1k20v8kc3vckapqf5nws30brnni-bash-4.4-p23/bin/bash"
/nix/store/ki845cvn6garxi6l7bdwjj3msm0hfzg5-git-sync-20151024
```
</p></details>
